BepInEx.Preloader.Core/Patching/BasePatcherProvider.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System.Collections.Generic;
+using System.Reflection;
+using BepInEx.Configuration;
+using BepInEx.Logging;
+
+namespace BepInEx.Preloader.Core.Patching;
+
+/// <summary>
+///     A patcher provider that can discover and load assemblies containing patcher plugins
+/// </summary>
+public abstract class BasePatcherProvider
```

Comment:
This much code duplication with regular plugin providers is not acceptable imo

Reply:
I investigated and found that the best I can do is to extract an interface, but not remove the duplication until the second pr which will clean the api. The reason is I need to unity the plugins metadata types before I can do this which starts to be a bit out of scope for this pr